### PR TITLE
cfg: move experimental options to main protocol cfg

### PR DESCRIPTION
**Pull request by orbitalturtle**
## Change Description

This PR simply moves the experimental custom message config options to the main protocol config, so that the dev build flag doesn't need to be set to use them. Closes issue #8536.

We've gotten multiple requests for this at LNDK -- people don't want to be running LND in dev mode in production. There hasn't been any discussion on the #8536 issue, so hoping to at least spur discussion on what needs to happen to make this change feasible. :)

Tested this with LNDK to confirm it works properly.

## Steps to Test
The config options set in `sample-lnd.conf` should be possible to set without the dev flag.

### CODE REVIEW

File: lncfg/protocol_integration.go
Comment (ellemouton) [inline lncfg/protocol_integration.go:20]: I think we'd want to leave this in & rather just remove the custom msg stuff out of the `ExperimentalProtocol` struct. We still want to leave the framework for adding experimental features.

**Review by ellemouton — COMMENTED** [review]
Nice - just left comment about implementation. 

Also just want to check offline with some folks to check if perhaps a better move would be to add an `experimental` build tag instead. Will get back to you here about the response.

**Comment by mrfelton** [comment]
> Also just want to check offline with some folks to check if perhaps a better move would be to add an experimental build tag instead. Will get back to you here about the response.

Ideally, these should be enabled in the default build to make these more accessible.

It would be amazing to get this into 0.18 to enable people to leverage these features without needing to create custom builds with special tags enabled. That would help further adoption of bolt 12 through lndk.

**Comment by guggero** [comment]
cc @saubyk

File: lncfg/protocol_experimental_off.go
Comment (ellemouton) [inline lncfg/protocol_experimental_off.go:13]: i still think we want to leave this struct here. Ie, leave the framework in place & just move the custom fields out

Comment (ellemouton) [reply lncfg/protocol_experimental_off.go:13]: if you "git blame", you can see that the framework was added independently of the custom message options

Comment (orbitalturtle) [reply lncfg/protocol_experimental_off.go:13]: Thanks for taking a look @ellemouton! That makes sense. I reworked to leave that framework in place and just move the "custom" protocol features out. Lmk if this is what you had in mind.

File: sample-lnd.conf
Comment (ellemouton) [inline sample-lnd.conf:1320]: should we maybe just note that a user can repeat the flags to set multiple bits? Else I think it comes across like they should specify the encoded bit vector

**Review by ellemouton — APPROVED** [review]
Code LGTM! one suggestion re the comments in the sample-conf file. 

Just need a product-ack from @saubyk  on this one too. 

Also might need to move the release note entry depending on if this is going in 0.18 or 0.18.1 - again up to @saubyk

**Comment by saubyk** [comment]
Trying to understand the need for this push given we are so close to wrapping up 0.18.
Assuming we have sophisticated users for LNDK, why can’t  they do custom build with make install and the tag? Is it because that it's a "dev" tag that it appears problematic?

**Comment by mrfelton** [comment]
> Trying to understand the need for this push given we are so close to wrapping up 0.18. Assuming we have sophisticated users for LNDK, why can’t they do custom build with make install and the tag?

I understand your concern about the timing of this push for 0.18. While LNDK users may tend to be technically proficient right now (partially because they need to be due to the current barriers to entry, like this one), Interest in Bolt 12 extends beyond advanced users, and LNDK is currently the primary path for enabling partial adoption for the vast majority of Lightning Network users who rely on LND.

I'd summarise as:

* **Reduced Barrier to Entry:** Currently, using these features requires compiling LND with a specific tag, a hurdle for anyone new to the system. By including it in the main configuration, users of all technical backgrounds can simply download the latest LND release (or use the official Lightning Labs Docker image, or spin up a setup using Polar) and install and connect LNDK. Removing the need for custom builds, custom docker images, and enabling more easy integration with tools like Polar would lower the barrier to entry and encourages wider adoption.

* **Potential Future Integration:** Including these options in the main configuration lays the groundwork for future feature integration. This could lead to a more streamlined process for utilizing advanced functionalities in the future, benefiting both new and experienced users.

* **Early Feedback and Informed Development:** Including these features now allows us to gather valuable user feedback and inform future development decisions related to Bolt 12 and LNDK integration. This can help ensure these features meet the needs of the broader user base.

While the 0.18 release cycle is nearing completion, the benefits of broader accessibility, smoother user experience, and early user feedback may outweigh the potential drawbacks of including this change now. It can significantly boost adoption and encourage more people to explore these features.

> Is it because that it's a "dev" tag that it appears problematic?

That is a problem. The `dev` tag is intended to be used for development builds. Many people would not want to run development builds in production, but  would want to use this specific feature.

**Comment by saubyk** [comment]
Tagged for 0.18. Gated on @carlaKC 's review

**Review by carlaKC — APPROVED** [review]
tACK, simple code move ✅

File: sample-lnd.conf
Comment (orbitalturtle) [reply sample-lnd.conf]: That makes sense, just improved those docs.

File: lncfg/protocol.go
Comment (yyforyongyu) [inline lncfg/protocol.go:139]: this can be removed

File: lncfg/protocol_experimental_on.go
Comment (yyforyongyu) [inline lncfg/protocol_experimental_on.go:3]: think the experiment on and off files can also be removed now?

Comment (orbitalturtle) [reply lncfg/protocol_experimental_on.go:3]: @ellemouton and I had a little convo about this up here: https://github.com/lightningnetwork/lnd/pull/8744#discussion_r1597945128

Leaving it in sounds like the right move to me because looking at the commit history, the experimental protocol struct has housed other protocol features in the past, so it could be useful for future experimental flags in the future.

**Pull request merged**
